agents/agent.py

```diff
@@ -20,6 +21,55 @@
     'agents.utils.logtools', 'logconfig.ini'
 )
 
+MAX_RETRIES = 3
+RETRY_DELAY = 5
```

Comment:
How about setting these values in agent's `config.ini`? This way, they can be set by the user in a flexible manner, instead of being 'hardcoded'.